modules/islandora_wos/islandora_wos.module

```diff
@@ -0,0 +1,139 @@
+      $object = menu_get_object('islandora_object', 2);
+      if ($object) {
+        $doi = islandora_badges_get_doi($object);
+        if ($doi) {
+          // Embed Web of Science.
+          // Set API endpoint URL.
+          // @TODO: Add to admin form.
+          $url = "https://ws.isiknowledge.com/cps/xrpc";
+
+          // Store XML Request in string, as I can't send an XML file by POST.
+          $input_xml = '<?xml version="1.0" encoding="UTF-8" ?>
+      <request xmlns="http://www.isinet.com/xrpc42" src="app.id=API Demo">
+        <fn name="LinksAMR.retrieve">
+          <list>
+            <!-- WHO IS REQUESTING -->
+            <map>   
+              <val name="username">' . $wos_username . '</val>       
+              <val name="password">' . $wos_password . '</val>            
+            </map>
+            <!-- WHAT IS REQUESTED -->
+            <map>
+              <list name="WOS">
+                <val>timesCited</val>
+                <val>citingArticlesURL</val>
+              </list>
+            </map>
+            <!-- LOOKUP DATA -->
+            <map>  
+              <map name="cite_1">
+                <val name="doi">' . $doi . '</val>
+              </map>
+            </map>
+          </list>
+        </fn>
+      </request>';
+
+          // Post the request and get results!
+
+          $result = drupal_http_request($url, array(
+            'headers' => array('Content-Type' => 'text/xml'),
+            'data' => $input_xml,
+            'method' => 'POST',
+            'timeout' => 10,
+            )
+          );
+
+          // Convert the response to an array and remove the CDATA tags.
+
+          $wos_result = simplexml_load_string($result->data, 'SimpleXMLElement', LIBXML_NOCDATA);
+
+          // Default - zero citations -- should this really be present?
+          $times_cited = 0;
+
+          // Default - citing_articles_url - no citing articles = empty array.
+          $citing_articles_url = array();
+
+          // Get the values into the variables.
+          foreach ($wos_result->fn->map->map->map->val as $val) {
+            switch ((string) $val['name']) {
+              case 'times_cited':
+                $times_cited = $val[0];
+                break;
+
+              case 'citing_articles_url':
+                // URL is child of <val name="citingArticlesURL"> under CDATA.
+                // We need to pull this out.
+                $citing_articles_url = $val[0];
+
+                break;
+
+            }
+          }
+          // If there is no error, return data.
+          if ($times_cited > 0) {
+            if (empty($result->error)) {
+              $badge_type = variable_get('islandora_wos_badgetype');
+              if ($badge_type == 'text') {
+                $to_render['content'] = '<div class="wos_badge"><a href="' . $citing_articles_url . '" target="_blank">Web of Science citations: ' . $times_cited . '</a></div>';
```

Comment:
Same deal as both of these as https://github.com/Islandora/islandora_altmetrics/pull/12/files#diff-7e73e0ed6b5ad8db5ee47b40668a515dR74 - would be far friendlier to themes if these were renderable arrays instead of hard-coded `<div>` elements.